File: framework/languages/java/e2e-kotlin/AutomaticTurnDispatch/Client/src/main/java/systems/zlink/e2e/kotlin/automaticturn/ClientApplication.java
package systems.zlink.e2e.kotlin.automaticturn;

public final class ClientApplication {
    private ClientApplication() {
    }

    public static void run(String... args) {
        String scenario = args.length > 0 ? args[0] : "all";
        if ("d2".equals(scenario) || "ATD-D2".equals(scenario) || "ATD-D3".equals(scenario)) {
            ClientScenario.runD2(scenario);
        } else if ("ATD-E3".equals(scenario)) {
            ClientScenario.runE3();
        } else {
            ClientScenario.run(scenario);
        }
        System.out.println("automatic-turn-dispatch kotlin e2e result=passed");
        System.exit(0);
    }
}

File: framework/languages/java/e2e-kotlin/DiscoveryRegistryHa/Client/src/main/kotlin/systems/zlink/e2e/kotlin/discoveryregistryha/client/ClientApplication.kt
package systems.zlink.e2e.kotlin.discoveryregistryha.client

import com.fasterxml.jackson.databind.ObjectMapper
import org.springframework.boot.ApplicationArguments
import org.springframework.boot.WebApplicationType
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.builder.SpringApplicationBuilder
import org.springframework.context.annotation.Bean
import systems.zlink.e2e.kotlin.discoveryregistryha.ClientOptions
import systems.zlink.e2e.kotlin.discoveryregistryha.ClientScenario

@SpringBootApplication(
    proxyBeanMethods = false,
    scanBasePackages = ["systems.zlink.e2e.kotlin.discoveryregistryha.client"],
)
class ClientApplication {
    companion object {
        @JvmStatic
        fun run(vararg args: String) {
            val context = SpringApplicationBuilder(ClientApplication::class.java)
                .web(WebApplicationType.NONE)
                .run(*args)
            try {
                context.getBean(ClientScenario::class.java).run()
                println("discovery-registry-ha kotlin e2e result=passed")
            } finally {
                context.close()
            }
        }
    }

    @Bean
    fun objectMapper(): ObjectMapper = ObjectMapper()

    @Bean
    fun clientOptions(args: ApplicationArguments): ClientOptions =
        ClientOptions.parse(args.sourceArgs)

    @Bean
    fun clientScenario(
        json: ObjectMapper,
        clientOptions: ClientOptions,
    ): ClientScenario =
        ClientScenario(json, clientOptions)
}
